kube-client/src/config/mod.rs

```diff
@@ -271,6 +273,45 @@ impl Config {
         })
     }
 
+    /// Override configuration based on environment variables
+    ///
+    /// This is only intended for use as a debugging aid, and the specific variables and their behaviour
+    /// should **not** be considered stable across releases.
+    ///
+    /// Currently, the following overrides are supported:
+    ///
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_USER`: A Kubernetes user to impersonate, for example: `system:serviceaccount:default:foo` will impersonate the `ServiceAccount` `foo` in the `Namespace` `default`
```

Comment:
maybe `KUBERNETES_IMPERSONATE_USER` here? we are hoping to find more alignment with `kubectl` who don't currently have an evar - but they could feasibly add this latter - and they couldn't add anything kube-rs branded.

if not `KUBE_RS_IMPERSONATE_USER` is ok